modules/social_features/social_tagging/src/SocialTaggingService.php

```diff
@@ -122,12 +133,29 @@ public function getChildren($category) {
     $options = [];
     // Fetch main categories.
     foreach ($this->termStorage->loadTree('social_tagging', $category, 1) as $category) {
-      $options[$category->tid] = $category->name;
+      $options[$category->tid] = $this->getTranslatedLabel($category->tid);
     }
     // Return array.
     return $options;
   }
 
+  /**
+   * Returns the translated label of the taxonomy.
+   *
+   * @param int $tid
+   *   The term id.
+   *
+   * @return string
+   *   The translated label.
+   */
+  public function getTranslatedLabel($tid) {
```

Comment:
This method might use some more defensive check(s). 

Where's its trying to get the taxonomy term `$taxonomy_term = $this->termStorage->load($tid);` I think we might want to check if the `$tid` is actually having some value.

Reply:
Hey @frankgraave, I have been thinking about it to make it more defensive, but I don't think it's needed. You could always ask Shipsters for their second opinion.

Reason is, I call the getTranslatedLabel on 4 occasions which are only in this file itself:
1.    line 115 and 135
foreach ($this->termStorage->loadTree('social_tagging', 0, 1) as $category) {
$options[$category->tid] = $category->name;
}

- It load the terms tree, if there would be no taxonomy terms it would not even enter the foreach loop.

2. line 205 and 224
      $current_term = $this->termStorage->load($term['target_id']);
      // Must be a valid Term.
      if (!$current_term instanceof TermInterface) {
        continue;
      }
      // Get current terms parents.
      $parents = $this->termStorage->loadParents($current_term->id());
      $parent = reset($parents);
      $category = $this->getTranslatedLabel($parent->id());

      $tree[$parent->id()]['tags'][$current_term->id()] = [
        'url' => $url->toString(),
        'name' => $this->getTranslatedLabel($current_term->id()),
      ];

- If the current_term would be empty, it would return right away not even entering my function. 
- For the parent, this buildHierarchy function is only being called when the tagging split is being set in the .module file, so it always has a tree, thus parent.

Reply:
Fair enough. Then I can agree it is not needed I guess!